components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayStartupListener.java

```diff
@@ -38,6 +39,7 @@
     private JMSTransportHandler jmsTransportHandlerForTrafficManager;
     private JMSTransportHandler jmsTransportHandlerForEventHub;
     private GatewayArtifactSynchronizerProperties gatewayArtifactSynchronizerProperties;
+    private boolean isAPIsDeployedInSyncMode = false;
```

Comment:
does this needed

Reply:
This is to verify whether all the APIs in DB are deployed in the gateway. This will become true only after every API is deployed correctly. If not `deployAPIsInSyncMode()` will be again triggered until each and every API is deployed. If all the APIs get deployed correctly then `isAPIsDeployedInSyncMode  ` becomes `true` and server will startup.